manager/state/raft/raft.go

```diff
@@ -115,6 +122,8 @@ type Node struct {
 	stopMu sync.RWMutex
 	// used for membership management checks
 	membershipLock sync.Mutex
+	// used to set Node
+	nodeLock sync.Mutex
```

Comment:
Not sure why we need this lock - doesn't `isMember` prevent the node from being used until it's set? Was the old code racy?